core/src/main/java/hudson/model/User.java

```diff
@@ -868,7 +868,7 @@ public void doConfigSubmit(StaplerRequest req, StaplerResponse rsp) throws IOExc
         for (UserPropertyDescriptor d : UserProperty.all()) {
             UserProperty p = getProperty(d.clazz);
 
-            JSONObject o = json.optJSONObject("userProperty" + (i++));
+            JSONObject o = json.optJSONObject("userProperty" + i++);
```

Comment:
IMO parens around pre/post increment seem useful for readability.

Reply:
As long as there are spaces around each + sign this doesn't seem too bad, and keeping the linter happy has other benefits like preventing further violations from being introduced. (I'm not introducing any enforcement in this PR, but I might try enabling Checkstyle's [`UnnecessaryParentheses`](https://checkstyle.sourceforge.io/config_coding.html#UnnecessaryParentheses) check in a future PR.)